### PR TITLE
Add 3.2.1.1 and 3.2.1.9 test cases.

### PR DESCRIPTION
**Pull request by nimanthag**
Signed-off-by: nimanthag <nimanthag@wso2.com>

### CODE REVIEW

File: product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCases.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
+package org.wso2.am.scenario.tests.api.secure.userRoles;
```
Comment (mushthaq33) [inline product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCases.java]: Missing license headers.

File: product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCases.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }
+    @DataProvider(name = "ScopeAndInValidRoleDataProvider")
+    public static Object[][] ValidRoleDataProvider() {
+
+        return new Object[][]{
+                {"everyone", ITEM_ADD},
+                {"admn", ORDER_ADD},
+                {"Internal/Craetor", ORDER_VIEW}
+        };
+    }
```
[the remark this answers is not in the document]
Comment (nimanthag) [reply product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCases.java]: Others are front end validation. Hence did the possible one with the API. Is there a way to do FE validation check?

File: product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCases.java (hunk 1 above)
Comment (nimanthag) [reply product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCases.java]: Changes added.

File: product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCases.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rmsamitha) [reply product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCases.java]: We can validate the response from jaggery api call, as similar to what is done as front-end validation. For example,
if role is not existing, we can check the "isRoleExist" element in response json object. i.e. {"error" : false, "isScopeExist" : "false", "isRoleExist" : false}.

**Pull request merged**